app/src/main/java/com/mybus/activity/SearchActivity.java

```diff
@@ -150,14 +168,23 @@ public void onBackPressed() {
     }
 
     @Override
+    public void onLocationGeocodingComplete(String address) {
+        geocodingComplete(address, mLastLocation);
+    }
+
+    @Override
     public void onAddressGeocodingComplete(LatLng location) {
+        geocodingComplete(mCurrentQuery, location);
+    }
+
+    private void geocodingComplete(String query, LatLng location) {
         cancelProgressDialog();
-        if (mCurrentQuery == null || location == null) {
+        if (query == null || location == null) {
```

Comment:
I would suggest you to use an if-then-else statement instead of a return. It will make it easier to understand.

if (query == null || location == null) {
   // Show no result found message
   Toast.makeText(this, R.string.toast_no_result_found, Toast.LENGTH_SHORT).show();            
} else {
   // Set geocoding result
   Intent intent = new Intent();
   ...
}